### ISSUE TITLE
Pointer being freed was not allocated

### ISSUE THREAD

**Issue by Jasonvdb** [lightningdevkit/ldk-swift#78]
Getting a crash on [this line in react-native-ldk](https://github.com/synonymdev/react-native-ldk/blob/c00cd726816db4841c3c823ebb505a1a43ce7257/lib/ios/Ldk.swift#L493). Seems to only happen when accidentally cooperatively closing a channel with a disconnected peer instead of force closing. Reproducible almost every time so let me know if there are any other details you need.

Adding a check in react-native-ldk to patch this so not too urgent.

```
example(14018,0x70000eebe000) malloc: *** error for object 0x60000012b300: pointer being freed was not allocated
example(14018,0x70000eebe000) malloc: *** set a breakpoint in malloc_error_break to debug
2022-10-05 12:19:07.023119+0200 example[14018:1418996] [javascript] [Error: Cannot begin shutdown while peer is disconnected or we're waiting on a monitor update, maybe force-close instead?]
example(14018,0x70000eebe000) malloc: *** error for object 0x60000012b300: pointer being freed was not allocated
CoreSimulator 857.7 - Device: iPhone 14 (C8BBB7A1-CC93-456D-A600-9CD38012F9FB) - Runtime: iOS 16.0 (20A360) - DeviceType: iPhone 14
```

![Screen Shot 2022-10-05 at 12 20 22](https://user-images.githubusercontent.com/5300488/194039387-a8233785-e364-4ce0-8f03-280d3584de0a.png)

![Screen Shot 2022-10-05 at 12 19 58](https://user-images.githubusercontent.com/5300488/194039408-15523f99-d667-467f-9bf6-76c20b4c5333.png)

**Comment by arik-so** [comment]
Thanks for this report! Unfortunately I'm a bit preoccupied this week and next so I can't look into it immediately. 

I know it's a bit big of an ask, but do you think you might be able to write a unit test in [LDKSwiftTests.swift](https://github.com/lightningdevkit/ldk-swift/blob/main/ci/LDKSwift/Tests/LDKSwiftTests/LDKSwiftTests.swift) that reproduces this issue? Thanks either way!

**Comment by tnull** [comment]
I believe this was fixed with #80?

**Comment by arik-so** [comment]
Yeah, and even if that didn't fix it, the followup related fixes from the reimplementation should have. Closing this for now, but feel free to reopen should the issue arise again.
